src/tools/miri/tests/pass/coerce_non_capture_closure_to_fn_ptr.rs

```diff
@@ -20,7 +22,7 @@ fn main() {
     BAR(44, 45);
     let bar: unsafe fn(i32, i32) = BAR;
     unsafe { bar(46, 47) };
-    let boo: &dyn Fn(i32, i32) = &BAR;
+    let boo: &dyn Fn(i32, i32) = addr_of!(BAR);
```

Comment:
These statics are not mutable, why is there a change needed here?

Reply:
It's a warning, not a hard error detected by the `static_ref` lint. Perhaps it would be better to add `#![allow(static_ref)]` instead of making a change. What do you think?

Reply:
We shouldn't warn when taking reference to immutable statics. This entire lint is only about `static mut`.

Reply:
You mentioned in this [comment](https://github.com/rust-lang/rust/pull/117556#issuecomment-1813053728) the two types of references: shared and mutable. I understand it like this:

```rust
// mutable reference
static mut X: i32 = 1;
let x = &X;

// shared reference
static Y: i32 = 2;
let y = &Y;
```

Do you mean?

```rust
static mut X: i32 = 1;
let x = &X;
let mut y = &X;
```

Reply:
> Do you mean this?

Yes I meant the second. The context is mutable statics. We are considering both shared and mutable references to these mutable statics.

Reply:
Second also creates two shared references, just the second local is mutable. @obeis comment you link means this:

```Rust
static mut X: i32 = 1;
let x = &X;
let y = &mut X;
```

The issue is, as the comment says, that while there is a mutation of `X` while the reference `x` lives, you have UB. This does not occur in this setting:

```Rust
static X: i32 = 1;
let x = &X;
```

Because `X` is never going to change.